lib/inferno/repositories/presets.rb

```diff
@@ -8,14 +8,20 @@ module Repositories
     # Repository that deals with persistence for the `Preset` entity.
     class Presets < InMemoryRepository
       def insert_from_file(path)
-        case path
-        when /\.json$/
-          preset_hash = JSON.parse(File.read(path))
-        when /\.erb$/
-          templated = ERB.new(File.read(path)).result
-          preset_hash = JSON.parse(templated)
+        raw_contents =
+          case path
+          when /\.json$/
+            File.read(path)
+          when /\.erb$/
+            ERB.new(File.read(path)).result
+          end
+
+        if Application['base_url'].start_with? 'https://inferno-qa.healthit.gov'
+          raw_contents.gsub!('https://inferno.healthit.gov', 'https://inferno-qa.healthit.gov')
```

Comment:
I feel like this opens the gateway to a lot of `if Application['base_url'] ...` statements. Inferno really needs some environment-based config built into the framework, but we already require `.env` files in test kits, so we can't use dotenv files to prepopulate a bunch of host-specific `ENV[...]` settings.

That's honestly an issue for another ticket though. Maybe test kit writers can be really good at putting all their environment-specific test kit things into a preset, and then we'll support something like `ENV['INFERNO_ALLOWED_PRESETS'] = "preset_id1,preset_id2"`.

Reply:
> I feel like this opens the gateway to a lot of if Application['base_url'] ... statements.

I don't know that I agree. This is to address a specific issue with presets.

- Locally, you generally want the preset to point to prod. So test kits hard code presets pointing at `inferno.healthit.gov`, i.e. something different than `INFERNO_HOST`.
- On prod, you want them to point to `inferno.healthit.gov`, i.e. the same thing as `INFERNO_HOST`
- On QA, you want them to point to `inferno-qa.healthit.gov`, i.e. the same thing as `INFERNO_HOST`

Right now I handle this by altering all of the presets in the deployment repo so that they use erb to dynamically determine the correct host. We could add another environment variable in order to make things work correctly locally, in prod, and in QA, but that would require all of the presets that point to the reference server to be rewritten in every test kit. Or we could do this and let all of the existing presets work as they are everywhere, and let people continue to create them in a very straightforward way.

> Inferno really needs some environment-based config built into the framework, but we already require .env files in test kits, so we can't use dotenv files to prepopulate a bunch of host-specific ENV[...] settings.

I'm not sure what you mean here. That is exactly what the `.env` files are for. But telling test authors that instead of just pointing their presets to `https://inferno.healthit.gov/reference-server/r4`, they need to turn their preset into an erb file and point things to `<%= ENV.fetch('INFERNO_REFERENCE_SERVER_HOST') %>/reference-server/r4` or whatever is not great, and is the sort of thing people would mess up.

Reply:
Yeah this works for our purposes. I'm just thinking from the perspective of a generic framework. If someone wants to deploy a test kit to their production site and staging site but use the same preset file, they need to duplicate `.env.production` with the core settings we already mandate, add their own custom env variables, and have their ERB preset use those. AFAIK dotenv files aren't composable so it can get ugly to maintain. I realize this is a low-reward aspirational feature though, devs can just offer two presets with clearly labelled titles.

Reply:
From my understanding of our design, hardcoding "inferno.healthit.gov" anywhere in Framework violates our design (forming a dependency from core to one platform's public hosting locations).  I believe hardcoding "inferno.healthit.gov" in Test Kit presets violates our design too (forming a dependency from a test kit and one possible platform, though not quite as bad). Whether its worth the effort to decouple these for this type of thing... seems like Steve is saying 'no'.

@Jammjammjamm  Did you consider putting this logic into `inferno-deployment`, so core doesn't have to know anything about those domains?  As long as you considered that and decided it wasn't worth it, then I'm ok with this approach all things considered.